File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let answer = 0;
    const vowels = ["A", "E", "I", "O", "U"];

    dfs("");

    function dfs(str) {
        if (str === word) return true;
        if (str.length === 5) return false;

        for (v of vowels) {
            str += v;
            answer++;

            if (dfs(str)) return true;

            str = str.substring(0, str.length - 1);
        }
    }

    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    vowels = ['A', 'E', 'I', 'O', 'U']
    weights = [781, 156, 31, 6, 1]
    cnt = 0

    for i, w in enumerate(word):
        cnt += vowels.index(w) * weights[i] + 1

    return cnt